### DISCUSSION TITLE
Allow all user fields to be public or private

### DISCUSSION THREAD

**Discussion by Aragur** [pocketbase/pocketbase#1348] [Q&A]
I'm currently implementing a user directory and I want to allow a user to select which fields of his account should be private (e.g., the address).
I noticed that setting the E-Mail to private will also hide it in each API call.
<img width="324" alt="Screen Shot 2022-12-23 at 00 05 09" src="https://user-images.githubusercontent.com/11004008/209239951-4cf99689-5d3e-4e6e-bdf3-4b7eeaaea4ff.png">
Is this also possible for other custom fields because I saw nothing in the UI nor the docs?

**Comment by ganigeorgiev** [comment]
No, the other fields don't have such option. Also note that "emailVisibility" field has a little more semantic than just hide/show since the user owner, admins and those with "Manage API Rule" access will be able to retrieve the email.

At the moment if you want to "hide" fields, you can move them to a separate collection and add `relation` type field pointing to it (eg. similar to "profile"). Another approach would be to [extend PocketBase](https://pocketbase.io/docs/use-as-framework/) and overwrite the [`OnRecord*Request`](https://pocketbase.io/docs/event-hooks/#record-api-hooks) hook responses, but this is more involving.
